### ISSUE TITLE
If you have errors doing "make repo-deps" read this!

### ISSUE THREAD

**Issue by g00dnatur3** [OpenZWave/python-openzwave#122]
You are probably running 64-bit linux and need to install the 32-bit libs:

Here are instructions for ubuntu:

`http://stackoverflow.com/questions/23182765/how-to-install-ia32-libs-in-ubuntu-14-04-lts-trusty-tahr`

**Comment by Dodobibi** [comment]
Thanks,

Cython work nice in 64-bit. There is no problem when installing it with "apt-get install cython" (or cython3 for python3.

(I'm needed only the lib, not the api, web, and manager and I build the lib in debian 64, python 3.4 (64), and cython 0.21.1 (64) succesfully. No need to install ia-32-libs.

Cheers.

**Comment by bibi21000** [comment]
As far as I know, using cython installed with pip always result in segfault. Maybe things have changed with the last release of python.
So if you want to use it, please report issues here : https://github.com/g00dnatur3/python-openzwave. 

@Dodobibi : was too hot in burgundy last night, so I read doc of cython again and again : 
 - we must call PyEval_InitThreads and use the with gil statement in callbacks like said here https://docs.python.org/dev/c-api/init.html#PyEval_InitThreads
There is a note for python 3 : Changed in version 3.2: This function cannot be called before Py_Initialize() anymore. 
Maybe we should call Py_Initialize() before.
Don't have the time to do it before monday, so if you want, you can try it and submit a pull request if it works for you.

@g00dnatur3 : like said too many times, please copy/paste errors when submitting issues.

**Comment by g00dnatur3** [comment]
@Dodobibi: The 32-bit library fixes errors not related to Cython. There are compile errors I received 

Please disregard my pip install comments, I edited it out.

I agree with how Cython is installed in make repo-deps.

But the 32-bit libraries did fix issues for me. Just install 64-bit ubuntu 15.04 and do a make repo-deps and you will see some compile errors.

**Comment by g00dnatur3** [comment]
It could be an issue to my specific hardware as it is an AMD laptop machine.

**Comment by bibi21000** [comment]
>The 32-bit library fixes errors not related to Cython. There are compile errors I received : 
where are they ???

>But the 32-bit libraries did fix issues for me. Just install 64-bit ubuntu 15.04 and do a make repo-deps and you will see some compile errors.
I can't see why they should be needed

>It could be an issue to my specific hardware as it is an AMD laptop machine.
I use a Toshiba Satellite C70D-A-10K (which hold an AMD A6 CPU) since 3 years

By the way, there is surely a problem with your computer, specially between keyboard and screen :)

**Comment by Dodobibi** [comment]
>was too hot in burgundy last night, so I read doc of cython again and again :

Was too hot in Franche-Comté also ! And I read the same article of you... :)

I don't think than the line .....__Pyx_PyBool_FromLong...... when compiling is a probleme. It's juste a c++ macro for converting long to bool values, but why it does not appear when compiling in 32-bit keep a mystery...

I will try Py_Initialize() this evening (I'm at work), but i can't reproduce the segfault when Py_InitThreads() is uncommented (python3), so it may be difficult, but I will try with python 2 et 3 and I will post the result.

Have a nice week end !

**Comment by g00dnatur3** [comment]
Here is the error:

```
pip install pylint
Downloading/unpacking pylint
  Downloading pylint-1.4.4-py2.py3-none-any.whl (428kB): 428kB downloaded
Downloading/unpacking logilab-common>=0.53.0 (from pylint)
  Downloading logilab-common-1.0.2.tar.gz (190kB): 190kB downloaded
  Running setup.py (path:/tmp/pip-build-zuRF6y/logilab-common/setup.py) egg_info for package logilab-common
    
    warning: no files found matching '*.py' under directory 'test/input'
    warning: no files found matching '*' under directory 'doc/html'
Requirement already satisfied (use --upgrade to upgrade): six in /usr/lib/python2.7/dist-packages (from pylint)
Downloading/unpacking astroid>=1.3.6 (from pylint)
  Downloading astroid-1.3.6-py2.py3-none-any.whl (182kB): 182kB downloaded
Requirement already satisfied (use --upgrade to upgrade): setuptools in /usr/lib/python2.7/dist-packages (from logilab-common>=0.53.0->pylint)
Installing collected packages: pylint, logilab-common, astroid
Compiling /tmp/pip-build-zuRF6y/pylint/pylint/test/functional/abstract_abc_methods.py ...
  File "/tmp/pip-build-zuRF6y/pylint/pylint/test/functional/abstract_abc_methods.py", line 6
    class Parent(object, metaclass=abc.ABCMeta):
                                  ^
SyntaxError: invalid syntax

Compiling /tmp/pip-build-zuRF6y/pylint/pylint/test/functional/abstract_class_instantiated_py2.py ...
  File "/tmp/pip-build-zuRF6y/pylint/pylint/test/functional/abstract_class_instantiated_py2.py", line 15
    class GoodClass(object, metaclass=abc.ABCMeta):
                                     ^
SyntaxError: invalid syntax

Compiling /tmp/pip-build-zuRF6y/pylint/pylint/test/functional/abstract_class_instantiated_py3.py ...
  File "/tmp/pip-build-zuRF6y/pylint/pylint/test/functional/abstract_class_instantiated_py3.py", line 14
    class GoodClass(object, metaclass=abc.ABCMeta):
                                     ^
SyntaxError: invalid syntax

Compiling /tmp/pip-build-zuRF6y/pylint/pylint/test/functional/abstract_method_py2.py ...
  File "/tmp/pip-build-zuRF6y/pylint/pylint/test/functional/abstract_method_py2.py", line 35
    class Structure(object, metaclass=abc.ABCMeta):
                                     ^
SyntaxError: invalid syntax

Compiling /tmp/pip-build-zuRF6y/pylint/pylint/test/functional/abstract_method_py3.py ...
  File "/tmp/pip-build-zuRF6y/pylint/pylint/test/functional/abstract_method_py3.py", line 35
    class Structure(object, metaclass=abc.ABCMeta):
                                     ^
SyntaxError: invalid syntax

Compiling /tmp/pip-build-zuRF6y/pylint/pylint/test/functional/class_members_py27.py ...
  File "/tmp/pip-build-zuRF6y/pylint/pylint/test/functional/class_members_py27.py", line 34
    class TestMetaclass(object, metaclass=ABCMeta):
                                         ^
SyntaxError: invalid syntax

Compiling /tmp/pip-build-zuRF6y/pylint/pylint/test/functional/class_members_py30.py ...
  File "/tmp/pip-build-zuRF6y/pylint/pylint/test/functional/class_members_py30.py", line 34
    class TestMetaclass(object, metaclass=ABCMeta):
                                         ^
SyntaxError: invalid syntax

Compiling /tmp/pip-build-zuRF6y/pylint/pylint/test/functional/no_name_in_module.py ...
  File "/tmp/pip-build-zuRF6y/pylint/pylint/test/functional/no_name_in_module.py", line 15
    print('hello world', file=sys.stdout)
                             ^
SyntaxError: invalid syntax

Compiling /tmp/pip-build-zuRF6y/pylint/pylint/test/functional/old_style_class_py27.py ...
  File "/tmp/pip-build-zuRF6y/pylint/pylint/test/functional/old_style_class_py27.py", line 10
    class NotOldStyle2(metaclass=type):
                                ^
SyntaxError: invalid syntax

Compiling /tmp/pip-build-zuRF6y/pylint/pylint/test/functional/raising_non_exception_py3.py ...
  File "/tmp/pip-build-zuRF6y/pylint/pylint/test/functional/raising_non_exception_py3.py", line 13
    raise Exc from missing # [raising-non-exception]
                 ^
SyntaxError: invalid syntax

Compiling /tmp/pip-build-zuRF6y/pylint/pylint/test/functional/unbalanced_tuple_unpacking_py30.py ...
  File "/tmp/pip-build-zuRF6y/pylint/pylint/test/functional/unbalanced_tuple_unpacking_py30.py", line 9
    first, second, *last = (1, 2, 3, 4)
                   ^
SyntaxError: invalid syntax

Compiling /tmp/pip-build-zuRF6y/pylint/pylint/test/functional/undefined_variable_py30.py ...
  File "/tmp/pip-build-zuRF6y/pylint/pylint/test/functional/undefined_variable_py30.py", line 8
    def test(self)->Undefined: # [undefined-variable]
                  ^
SyntaxError: invalid syntax

Compiling /tmp/pip-build-zuRF6y/pylint/pylint/test/functional/yield_outside_func.py ...
  File "/tmp/pip-build-zuRF6y/pylint/pylint/test/functional/yield_outside_func.py", line 2
    yield 1  # [yield-outside-function]
SyntaxError: 'yield' outside function

/tmp/pip-build-zuRF6y/pylint/pylint/test/input/func_assert_2uple.py:4: SyntaxWarning: assertion is always true, perhaps remove parentheses?
  assert (1 == 1, 2 == 2), "no error"
/tmp/pip-build-zuRF6y/pylint/pylint/test/input/func_assert_2uple.py:5: SyntaxWarning: assertion is always true, perhaps remove parentheses?
  assert (1 == 1, 2 == 2) #this should generate a warning
/tmp/pip-build-zuRF6y/pylint/pylint/test/input/func_assert_2uple.py:7: SyntaxWarning: assertion is always true, perhaps remove parentheses?
  assert (1 == 1, ), "no error"
/tmp/pip-build-zuRF6y/pylint/pylint/test/input/func_assert_2uple.py:8: SyntaxWarning: assertion is always true, perhaps remove parentheses?
  assert (1 == 1, )
/tmp/pip-build-zuRF6y/pylint/pylint/test/input/func_assert_2uple.py:9: SyntaxWarning: assertion is always true, perhaps remove parentheses?
  assert (1 == 1, 2 == 2, 3 == 5), "no error"
/tmp/pip-build-zuRF6y/pylint/pylint/test/input/func_assert_2uple.py:11: SyntaxWarning: assertion is always true, perhaps remove parentheses?
  assert (True, 'error msg') #this should generate a warning
Compiling /tmp/pip-build-zuRF6y/pylint/pylint/test/input/func_bad_exception_context_py30.py ...
  File "/tmp/pip-build-zuRF6y/pylint/pylint/test/input/func_bad_exception_context_py30.py", line 14
    raise IndexError from 1
                        ^
SyntaxError: invalid syntax

Compiling /tmp/pip-build-zuRF6y/pylint/pylint/test/input/func_continue_not_in_loop.py ...
  File "/tmp/pip-build-zuRF6y/pylint/pylint/test/input/func_continue_not_in_loop.py", line 8
    continue
SyntaxError: 'continue' not properly in loop

Compiling /tmp/pip-build-zuRF6y/pylint/pylint/test/input/func_e0108.py ...
SyntaxError: duplicate argument '_' in function definition (func_e0108.py, line 4)

Compiling /tmp/pip-build-zuRF6y/pylint/pylint/test/input/func_exec_used_py30.py ...
  File "/tmp/pip-build-zuRF6y/pylint/pylint/test/input/func_exec_used_py30.py", line 6
    exec('a = 1', globals={})
                         ^
SyntaxError: invalid syntax

Compiling /tmp/pip-build-zuRF6y/pylint/pylint/test/input/func_keyword_repeat.py ...
  File "/tmp/pip-build-zuRF6y/pylint/pylint/test/input/func_keyword_repeat.py", line 8
    function_default_arg(two=5, two=7)
SyntaxError: keyword argument repeated

Compiling /tmp/pip-build-zuRF6y/pylint/pylint/test/input/func_kwoa_py30.py ...
  File "/tmp/pip-build-zuRF6y/pylint/pylint/test/input/func_kwoa_py30.py", line 3
    def function(*, foo):
                  ^
SyntaxError: invalid syntax

Compiling /tmp/pip-build-zuRF6y/pylint/pylint/test/input/func_noerror_mcs_attr_access.py ...
  File "/tmp/pip-build-zuRF6y/pylint/pylint/test/input/func_noerror_mcs_attr_access.py", line 14
    class Test(object, metaclass=Meta):
                                ^
SyntaxError: invalid syntax

Compiling /tmp/pip-build-zuRF6y/pylint/pylint/test/input/func_noerror_unused_variable_py30.py ...
  File "/tmp/pip-build-zuRF6y/pylint/pylint/test/input/func_noerror_unused_variable_py30.py", line 12
    nonlocal attr
                ^
SyntaxError: invalid syntax

Compiling /tmp/pip-build-zuRF6y/pylint/pylint/test/input/func_return_outside_func.py ...
  File "/tmp/pip-build-zuRF6y/pylint/pylint/test/input/func_return_outside_func.py", line 3
    return
SyntaxError: 'return' outside function

Compiling /tmp/pip-build-zuRF6y/pylint/pylint/test/input/func_return_yield_mix_py_33.py ...
SyntaxError: 'return' with argument inside generator (func_return_yield_mix_py_33.py, line 8)

Compiling /tmp/pip-build-zuRF6y/pylint/pylint/test/input/func_syntax_error.py ...
  File "/tmp/pip-build-zuRF6y/pylint/pylint/test/input/func_syntax_error.py", line 1
    def toto
           ^
SyntaxError: invalid syntax

Compiling /tmp/pip-build-zuRF6y/pylint/pylint/test/input/func_undefined_metaclass_var_py30.py ...
  File "/tmp/pip-build-zuRF6y/pylint/pylint/test/input/func_undefined_metaclass_var_py30.py", line 8
    class Bad(metaclass=ABCMet):
                       ^
SyntaxError: invalid syntax

Compiling /tmp/pip-build-zuRF6y/pylint/pylint/test/input/func_unused_import_py30.py ...
  File "/tmp/pip-build-zuRF6y/pylint/pylint/test/input/func_unused_import_py30.py", line 10
    class Meta(metaclass=abc.ABCMeta):
                        ^
SyntaxError: invalid syntax

Compiling /tmp/pip-build-zuRF6y/pylint/pylint/test/input/func_used_before_assignment_py30.py ...
  File "/tmp/pip-build-zuRF6y/pylint/pylint/test/input/func_used_before_assignment_py30.py", line 10
    nonlocal cnt
               ^
SyntaxError: invalid syntax

Compiling /tmp/pip-build-zuRF6y/pylint/pylint/test/input/func_w0705.py ...
  File "/tmp/pip-build-zuRF6y/pylint/pylint/test/input/func_w0705.py", line 28
    __revision__ += 1
SyntaxError: default 'except:' must be last

Compiling /tmp/pip-build-zuRF6y/pylint/pylint/test/input/syntax_error.py ...
Sorry: IndentationError: expected an indented block (syntax_error.py, line 2)
  Running setup.py install for logilab-common
    changing mode of build/scripts-2.7/pytest from 644 to 755
    Skipping installation of /usr/local/lib/python2.7/dist-packages/logilab/__init__.py (namespace package)
    
    warning: no files found matching '*.py' under directory 'test/input'
    warning: no files found matching '*' under directory 'doc/html'
    Installing /usr/local/lib/python2.7/dist-packages/logilab_common-1.0.2-nspkg.pth
    changing mode of /usr/local/bin/pytest to 755
Compiling /tmp/pip-build-zuRF6y/astroid/astroid/tests/testdata/python3/data/module.py ...
  File "/tmp/pip-build-zuRF6y/astroid/astroid/tests/testdata/python3/data/module.py", line 55
    print('yo', end=' ')
                   ^
SyntaxError: invalid syntax

Compiling /tmp/pip-build-zuRF6y/astroid/astroid/tests/testdata/python3/data/module2.py ...
  File "/tmp/pip-build-zuRF6y/astroid/astroid/tests/testdata/python3/data/module2.py", line 100
    print('bonjour', file=stream)
                         ^
SyntaxError: invalid syntax

Successfully installed pylint logilab-common astroid
```

**Comment by g00dnatur3** [comment]
Steps to reproduce:

1) install ubuntu 15.04 64-bit
2) sudo apt-get update
3) sudo apt-get upgrade
4) run theubuntu software updater
5) follow the directions in INSTALL_REPO.txt file

To fix error this I installed the 32-bit libs as described above

@bibi21000: I think the real problem is in your character & between you ears. Seriously, what type of person implicitly calls another person an idiot whose intentions are good. I am only trying to help out other developers who come across the same issues i have. 

Also, when you start -> stop-> start the zwave network it still doesnt work. it just doesnt crash now but it has database errors. You would know this if you actually verified a bug you supposedly "fixed".

Oh wait, you want me to post you the error message? Why dont you actually run the code you write instead of accusing others of being stupid.

**Comment by bibi21000** [comment]
> @bibi21000: I think the real problem is in your character & between your ears.

I'm sure I'm the problem too :D 

> Seriously, what type of person implicitly calls another person an idiot whose intentions are good. I am only trying to help out other developers who come across the same issues i have.

You really think you're helping people telling them install cython via pip like in your original post ? As said in README, I always get a segfault using that method. Of course, things change ... and maybe that should work now. That's why I've put a link to your fork for  this issue.

> Also, when you start -> stop-> start the zwave network it still doesnt work. it just doesnt crash now but it has database errors. You would know this if you actually verified a bug you supposedly "fixed".

You spoke about that issue https://github.com/OpenZWave/python-openzwave/issues/117 ?
It should be fixed and yes I ran it, I've added a test for non-regression too : Look at https://github.com/OpenZWave/python-openzwave/blob/952b39cb545854caad52e02b0857f11f1876c634/tests/api/test_network_start.py#L113

Maybe it shouldn't work for you, that's possible. As you said I've got a problem between my ears :D

One more time : like said in the doc and in others issues, it's a bad idea to stop/start the nework. It should be better to better to destroy it and create a new one (like said in doc). 
Why ? : I'm pretty sure that you will get a segfault if you do it many times ( don't know if it's because of the c++ lib, cython, ptyhon, ...). 
If you use this lib in a real world, you surely seen that the complete startup of the network take from minutes to one hour. So it's not really problematic to waste some seconds destroying the network object and recreating it.

> Oh wait, you want me to post you the error message? 

NO NO NO, I don't want anything from you. 

I just tell you so many times that if you want help (look at all your closed issues if you don't remenber), you must at least provide some informations (look at README again).

If I try to resume this issue : 
- at post 4 we got the information that you ran on Ubuntu 15.04 64bits
- at post 8 we've got the message

And finally, please do not edit your posts allt the time, it's really hard to follow.

By the way, I'm thinking I waste my time again explaining you same things again and again ....
I've spent 30 minutes writing a response to this post instead of reponding a real problem

In the future, all your malformated issues will be closed without any comment. And if you're tempted to spam this issues, you will simply be banned from this repository.
The only way to you to get help will be the google group. Maybe you will find someone asking you all the needed informations to open an issue.

For the others, if you're really interresting in helping this project, please report your problem with segfault to the cython-users group here : https://groups.google.com/forum/#!topic/cython-users/mRsviGuCFOk

It's time for week-end now !!!

**Comment by g00dnatur3** [comment]
Why are you still rambling on about Cythom. I already agreed with on your Cython install & observations.

This thread was only meant to say "hey if you get compile errors when running make repo-deps" try installing 32bit libs.

you had to deny my observation of comple errors and implicitly call me an idiot.

And I took 2 hours to redo everything just so I could get the exact compile errors for you to see.

Have a nice weekend
